packages/button/src/Button/Button.tsx

```diff
@@ -15,74 +18,74 @@ import { getClassName } from './Button.styles';
 /**
  * Buttons allow users to take actions, and make choices, with a single tap.
  */
-export const Button = React.forwardRef(function Button(
-  {
-    variant = Variant.Default,
-    size = Size.Default,
-    darkMode: darkModeProp,
-    'data-lgid': dataLgId = LGIDS_BUTTON.root,
-    baseFontSize = BaseFontSize.Body1,
-    disabled = false,
-    onClick,
-    leftGlyph,
-    rightGlyph,
-    children,
-    className,
-    as,
-    type,
-    isLoading = false,
-    loadingIndicator,
-    loadingText,
-    ...rest
-  }: BoxProps<'button', ButtonProps>,
-  forwardRef,
-) {
-  const { darkMode } = useDarkMode(darkModeProp);
+export const Button = InferredPolymorphic<ButtonProps, 'button'>(
```

Comment:
We should export a prop type that is also wrapped in InferredPolymorphic. Similar to [splitButton](https://github.com/mongodb/leafygreen-ui/blob/main/packages/split-button/src/SplitButton/SplitButton.types.ts#L146). This way, if consumers need to extend the type, it will include the inferred polymorphic props.